Tools/ardupilotwaf/configuration_helper.py

```diff
@@ -0,0 +1,78 @@
+"""
+WAF Tool that helps on configuration parameter, creating the AP_Config
+header file.
+
+This tool needs compiler_c and compiler_cxx to be loaded, make sure
+you load them before this tool.
+
+Example::
+    def configure(cfg):
+        cfg.load('configuration_helper')
+"""
+
+def configure(cfg):
+    cfg.check_cxx(fragment='''
+                  #include <cmath>
+
+                  int main() {
+                    return std::isfinite(1.0f);
+                  }''',
+                  define_name="HAVE_CMATH_ISFINITE",
+                  mandatory=False)
+
+    cfg.check_cxx(fragment='''
+                  #include <cmath>
+
+                  int main() {
+                    return std::isinf(1.0f);
+                  }''',
+                  define_name="HAVE_CMATH_ISINF",
+                  mandatory=False)
+
+    cfg.check_cxx(fragment='''
+                  #include <cmath>
+
+                  int main() {
+                    return std::isnan(1.0f);
+                  }''',
+                  define_name="HAVE_CMATH_ISNAN",
+                  mandatory=False)
+
+    cfg.check_cxx(fragment='''
+                  #include <math.h>
+                  #include <cmath>
+
+                  using std::isfinite;
+
+                  int main() {
+                    return isfinite((double)1);
+                  }''',
+                  define_name="NEED_CMATH_ISFINITE_STD_NAMESPACE",
+                  cxxflags="-std=gnu++11",
```

Comment:
@guludo I'll add for the previous ones

@OXINARF We use `-std=gnu++11` to compile the project. In this part, `waf` will try to build this pieces of code and generate flags based on the result, we need to explicitly use this flag here since `check_cxx` don't use it by default.